listenbrainz_spark/path.py

```diff
@@ -1,12 +1,40 @@
 import os
 
+from listenbrainz_spark import config
+
+# Parent directory containing listens and data driven from ListenBrainz.
+LISTENBRAINZ_DATA_DIRECTORY = os.path.join('/', 'data', 'listenbrainz')
+# Directory containing similar artist relation.
+# (This is a temporary path till incremental dumps for similar artists are prepared)
+SIMILAR_ARTIST_DIR = os.path.join(LISTENBRAINZ_DATA_DIRECTORY, 'similar_artists')
+
+
 # Parent directory containing data used in and generated by recommendation engine.
 RECOMMENDATION_PARENT_DIR = os.path.join('/', 'recommendation')
 # Directory containing dataframes to be used in generating recommendations (not necessarily).
-DATAFRAME_DIR = os.path.join('/', 'recommendation', 'dataframe')
-# Directory containing models to be used for generating recommendations.
-MODEL_DIR = os.path.join('/', 'recommendation', 'model')
+DATAFRAME_DIR = os.path.join(RECOMMENDATION_PARENT_DIR, 'dataframe')
+# Directory containing model metadata and the real models to be used for generating recommendations.
+MODEL_DIR = os.path.join(RECOMMENDATION_PARENT_DIR, 'model')
 # Directory containing candidate sets to be used for generating recommendations.
-CANDIDATE_SET_DIR = os.path.join('/', 'recommendation', 'candidate_set')
+CANDIDATE_SET_DIR = os.path.join(RECOMMENDATION_PARENT_DIR, 'candidate_set')
 # Directory containing RDD checkpoints to break lineage while using iterative algorithms.
 CHECKPOINT_DIR = os.path.join('/', 'checkpoint')
+# Directory to save best models
+DATA_DIR = os.path.join(MODEL_DIR, 'data')
+
+# Absolute path to dataframes used in processing raw data/listens.
+USERS_DATAFRAME_PATH = config.HDFS_CLUSTER_URI + DATAFRAME_DIR + '/' + 'users_df.parquet'
```

Comment:
any particular reason for us to not use os.path.join here?

Reply:
and in the lines that follow

Reply:
https://stackoverflow.com/questions/1945920/why-doesnt-os-path-join-work-in-this-case